pkg/pubsub/config.go

```diff
@@ -0,0 +1,55 @@
+package pubsub
+
+import (
+	"fmt"
+	"time"
+
+	cbuilder "github.com/scribd/go-sdk/internal/pkg/configuration/builder"
+)
+
+type (
+	Config struct {
+		Kafka Kafka `mapstructure:"kafka"`
+	}
+
+	Publisher struct {
+		MaxAttempts  int           `mapstructure:"max_attempts"`
+		WriteTimeout time.Duration `mapstructure:"write_timeout"`
+		Topic        string        `mapstructure:"topic"`
+		Enabled      bool          `mapstructure:"enabled"`
+	}
+
+	Subscriber struct {
+		Topic   string `mapstructure:"topic"`
+		GroupId string `mapstructure:"group_id"`
+		Enabled bool   `mapstructure:"enabled"`
+	}
+
+	Kafka struct {
+		BrokerUrls             []string   `mapstructure:"broker_urls"`
+		ClientId               string     `mapstructure:"client_id"`
+		Cert                   string     `mapstructure:"cert_pem"`
+		CertKey                string     `mapstructure:"cert_pem_key"`
+		SecurityProtocol       string     `mapstructure:"security_protocol"`
+		Publisher              Publisher  `mapstructure:"publisher"`
+		Subscriber             Subscriber `mapstructure:"subscriber"`
+		SSLVerificationEnabled bool       `mapstructure:"ssl_verification_enabled"`
+	}
+)
+
+// NewConfig returns a new Config instance.
+func NewConfig() (*Config, error) {
+	config := &Config{}
+	viperBuilder := cbuilder.New("pubsub")
+
+	vConf, err := viperBuilder.Build()
+	if err != nil {
+		return config, err
+	}
+
+	if err = vConf.Unmarshal(config); err != nil {
+		return config, fmt.Errorf("unable to decode into struct: %s", err.Error())
```

Comment:
nitpick: Shouldn't the message start with a capital letter?

Reply:
@terranisu It is [recommended](https://github.com/golang/go/wiki/CodeReviewComments#error-strings) to not use capital letters in the error messages

>Error strings should not be capitalized (unless beginning with proper nouns or acronyms) or end with punctuation, since they are usually printed following other context. That is, use fmt.Errorf("something bad") not fmt.Errorf("Something bad"), so that log.Printf("Reading %s: %v", filename, err) formats without a spurious capital letter mid-message. This does not apply to logging, which is implicitly line-oriented and not combined inside other messages.